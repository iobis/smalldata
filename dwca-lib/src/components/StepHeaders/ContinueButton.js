import PropTypes from 'prop-types'
import React from 'react'
import { useTranslation } from 'react-i18next'

export default function ContinueButton({ name, value, onClick }) {
  const { t } = useTranslation()
  const label = t('common.continueTo') + ' ' + value.toLowerCase()

  return <button className="continue-button button is-info" name={name} onClick={onClick}>{label}</button>
}

ContinueButton.propTypes = {
  name:    PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  value:   PropTypes.string.isRequired
}
