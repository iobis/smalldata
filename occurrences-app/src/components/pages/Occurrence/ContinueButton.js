import PropTypes from 'prop-types'
import React from 'react'
import { useTranslation } from 'react-i18next'

export default function ContinueButton({ name, value, onClick }) {
  const { t } = useTranslation()

  return (
    <div>
      <button
        className="button is-info"
        name={name}
        onClick={onClick}>
        {t('occurrenceForm.stepContinue')}{value}
      </button>
    </div>
  )
}

ContinueButton.propTypes = {
  name:    PropTypes.string,
  onClick: PropTypes.func.isRequired,
  value:   PropTypes.string.isRequired
}
