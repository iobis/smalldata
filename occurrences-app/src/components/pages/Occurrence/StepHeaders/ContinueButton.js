import PropTypes from 'prop-types'
import React from 'react'
import { useTranslation } from 'react-i18next'

export default function ContinueButton({ name, value, onClick }) {
  const { t } = useTranslation()
  const label = t('occurrenceForm.stepContinue') + ' ' + value.toLowerCase()

  return <button className="button is-info" name={name} onClick={onClick}>{label}</button>
}

ContinueButton.propTypes = {
  name:    PropTypes.string,
  onClick: PropTypes.func.isRequired,
  value:   PropTypes.string.isRequired
}
