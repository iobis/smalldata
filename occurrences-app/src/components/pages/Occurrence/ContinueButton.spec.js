import ContinueButton from './ContinueButton'
import React from 'react'
import { mount } from 'enzyme'

describe('ContinueButton', () => {
  it('renders correctly', () => {
    expect(mount(createComponent())).toMatchSnapshot()
  })

  it('renders a given value correctly', () => {
    const wrapper = mount(createComponent({ value: 'Go to next page' }))
    expect(wrapper.find('button').text()).toBe('Go to next page')
  })

  it('applies given CSS classes to wrapper correctly', () => {
    const wrapper = mount(createComponent({ wrapperClassName: 'my-own-class' }))
    expect(wrapper.find('div').at(0).hasClass('my-own-class')).toBe(true)
  })

  it('calls nextStepHandler', () => {
    const nextStepHandler = jest.fn()
    const wrapper = mount(createComponent({ nextStepHandler: nextStepHandler }))
    wrapper.find('button').at(0).simulate('click')

    expect(nextStepHandler).toHaveBeenCalledTimes(1)
  })
})

function createComponent(props) {
  const defaultProps = {
    name:             'continue',
    nextStepHandler:  jest.fn(),
    value:            'Continue',
    wrapperClassName: 'columns'
  }
  return <ContinueButton {...defaultProps} {...props}/>
}
