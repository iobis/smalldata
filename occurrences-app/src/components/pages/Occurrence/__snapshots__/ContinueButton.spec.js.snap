// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContinueButton renders correctly 1`] = `
<ContinueButton
  name="continue"
  onClick={[MockFunction]}
  value="Continue"
>
  <div>
    <button
      className="button is-info"
      name="continue"
      onClick={[MockFunction]}
    >
      occurrenceForm.stepContinue
      Continue
    </button>
  </div>
</ContinueButton>
`;
